cfblog/models.py

```diff
@@ -0,0 +1,146 @@
+    status = models.IntegerField(_('status'), choices=STATUS_CHOICES, default=1)
+    publish = models.DateTimeField(_('publish'), default=datetime.datetime.now)
+
+    class Meta:
+        abstract = True
+
+    @property
+    def is_public(self):
+        return self.status == 2
+
+    def publish_cms_content(self):
+        self.public_data = self.auth_data
+        self.status = 2
+        self.save()
+
+
+class CmsPage(BaseCmsPage):
+    url = models.CharField(_('url path'),
+                           max_length=255, db_index=True, unique=True, validators=[validate_url_path])
+
+    @cacheable('cms_page_{id}')
+    def html(self):
+        return get_public_data(self)
+
+    @stale_cache('cms_page_{id}')
+    def save(self, *args, **kwargs):
+        super(CmsPage, self).save(*args, **kwargs)
+
+    def clean(self):
+        from .views import cms_page_index
+        resolver_math = resolve(self.url)
+        if resolver_math.func is cms_page_index and not self.template:
+            raise ValidationError(_(
+                'Template cannot be empty for new urls'
+            ))
+
+        if resolver_math.func is not cms_page_index and self.template:
+            raise ValidationError(_(
+                'Template is not required for existing urls'
+            ))
+        super(CmsPage, self).clean()
+
+
+class CmsBlogCategory(models.Model):
+    """Category model."""
+    LEVEL_CHOICES = (
+        (1, 1),
+        (2, 2),
+    )
+    title = models.CharField(_('title'), max_length=100)
+    description = models.TextField(_('description'), blank=True)
+    slug = models.SlugField(_('slug'))
+    parent = models.ForeignKey('self', blank=True, null=True)
+    level = models.IntegerField(_('level'), choices=LEVEL_CHOICES)
+    is_active = models.BooleanField(_('is_active'), default=True)
+    objects = CmsBlogCategoryManager()
+
+    class Meta:
+        verbose_name_plural = _('cms blog page categories')
+        ordering = ('title',)
+
+    def __unicode__(self):
+        return self.title
+
+
+class CmsBlogPost(BaseCmsPage):
+    title = models.CharField(_('title'), max_length=200)
+    author = models.ForeignKey(settings.AUTH_USER_MODEL, blank=True, null=True)
+    category = models.ForeignKey('CmsBlogCategory')
+    tags = TagField()
+    slug = models.SlugField(_('slug'), max_length=70)
+    body = models.TextField(_('body'), null=True, blank=True)
+    tease = models.TextField(_('tease'), blank=True,
+                             help_text=_('Concise text suggested. Does not appear in RSS feed.'))
+    keywords = models.TextField(_('keywords'), blank=True,
+                                help_text=_('List of keywords, relevant for SEO'))
+    # TODO: Include S3ImageField
+    # thumbnail = ImageWithThumbsField(upload_to=get_image_path, sizes=(('s', 500, 500), ('r', 450, 600), ('t', 200, 260)), null=True, blank=True)
+    infographic = models.ImageField(upload_to=get_image_path, null=True, blank=True)
```

Comment:
infographic, pdf, allow_comments, sub_category, related_posts, votes to be removed. Keep notes.